src/Package/Impl/Utilities/ProjectUtilities.cs

```diff
@@ -42,5 +52,95 @@ public static class ProjectUtilities {
 
             return result;
         }
+
+        public static EnvDTE.Project GetActiveProject() {
+            DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
+            var projects = (Array)dte.ActiveSolutionProjects;
+            if (projects != null && projects.Length == 1) {
+                return (EnvDTE.Project)projects.GetValue(0);
+            }
+            return null;
+        }
+
+        public static void AddNewItem(string templateName, string name, string extension, string destinationPath) {
+            var project = GetActiveProject();
+            if (project != null) {
+                DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
+                var solution = (Solution2)dte.Solution;
+
+                // Construct name of the compressed template
+                var compressedTemplateName = Path.ChangeExtension(templateName, "zip");
+                var templatePath = Path.Combine(GetProjectItemTemplatesFolder(), compressedTemplateName);
+
+                // We will be extracting template contents into a temp folder
+                var uncompressedTemplateFolder = Path.Combine(Path.GetTempPath(), templateName);
+                var uncompressedTemplateName = Path.ChangeExtension(compressedTemplateName, "vstemplate");
+                var tempTemplateFile = Path.Combine(uncompressedTemplateFolder, uncompressedTemplateName);
+
+                // Extract template files overwriting any existing ones
+                using (ZipArchive zip = ZipFile.OpenRead(templatePath)) {
+                    foreach (ZipArchiveEntry entry in zip.Entries) {
+                        string destFilePath = Path.Combine(uncompressedTemplateFolder, entry.FullName);
+                        if (!string.IsNullOrEmpty(entry.Name)) {
+                            entry.ExtractToFile(destFilePath, true);
+                        } else {
+                            Directory.CreateDirectory(Path.GetDirectoryName(destFilePath));
+                        }
+                    }
+
+                    // Given path to the project or a folder in it, generate unique file name
+                    var fileName = GetUniqueFileName(destinationPath, name, extension);
+
+                    // Locate folder in the project
+                    var projectFolder = Path.GetDirectoryName(project.FullName);
+                    if (destinationPath.StartsWith(projectFolder, StringComparison.OrdinalIgnoreCase)) {
+                        ProjectItems projectItems = project.ProjectItems;
```

Comment:
Don't change ProjectItems explicitly - file system should handle the change automatically